crates/wasi/src/lib.rs

```diff
@@ -82,3 +82,45 @@ pub mod snapshots {
 }
 }
 }
+
+/// Exit the process with a conventional OS error code as long as Wasmtime
+/// understands the error. If the error is not an `I32Exit` or `Trap`, return
+/// the error back to the caller for it to decide what to do.
+///
+/// Note: this function is designed for CLI usage of Wasmtime; embedders of
+/// Wasmtime may want different error handling.
```

Comment:
I'd reword "CLI usage" to something like its "for usage where it is acceptable for wasmtime failures to terminate the parent process, such as in the wasmtime CLI", and say something like "not suitable for use in multi-tenant embeddings" as a more obvious warning? Could we hide the presence of this function behind an off-by-default cargo feature, to give more assurance to multi-tenant embeddings that they are safe from this?